tools/src/main/python/dlpx/virtualization/_internal/commands/build.py

```diff
@@ -145,6 +145,10 @@ def build(plugin_config,
     # Copy everything from the source directory into the build directory.
     file_util.clean_copy(src_dir, build_src_dir)
 
+    # compile the copied src directory. We compile here, as symlinks are also copied
+    # as folder and can be compiled without compiling the symlink source folder.
```

Comment:
we follow the multiline comment syntax of having a `#` before and after. Also we try to do full sentences so make sure to capitalize first words for comments.

The comment is somewhat confusing to me though. or maybe i just don't understand why this was the solution. I don't think the description of the PR was updated? why are we compiling here only to install other dependencies only to compile again?

Reply:
Updated the comment.

Also, the reason to go for this solution

- Before creating the zip file, we do compile the full build folder. But we do not use legacy=true due to which the .pyc files are not created within the same folder (created in __pycache__ folder in local cache directory) as .py files
- The above causes missing .pyc files for all the .py files present in symlinks folder.

To sort out the issue, we could have compiled the full build folder using  legacy=true which caused another issues related to protobuf version mismatch between docker engine(protobuf=3.19.0) and dvp-api(3.6.1).

So, we went via this approach where we first compile the build folder (Without any dependency in it), to make sure all pyc files are generated for .py files only in src folder.

I hope the above will help.